src/Symfony/Component/Serializer/Exception/NotNormalizableValueException.php

```diff
@@ -16,4 +16,45 @@
  */
 class NotNormalizableValueException extends UnexpectedValueException
 {
+    private $currentType = null;
+    private $expectedTypes = null;
+    private $path = null;
+    private $useMessageForUser = null;
+
+    /**
+     * @param bool $useMessageForUser If the message passed to this exception is something that can be shown
+     *                                safely to your user. In other words, avoid catching other exceptions and
+     *                                passing their message directly to this class.
+     */
+    public static function create(string $message, $data, array $expectedTypes, string $path = null, bool $useMessageForUser = false, int $code = 0, \Throwable $previous = null): self
+    {
+        $self = new self($message, $code, $previous);
+
+        $self->currentType = get_debug_type($data);
+        $self->expectedTypes = $expectedTypes;
+        $self->path = $path;
+        $self->useMessageForUser = $useMessageForUser;
+
+        return $self;
+    }
+
+    public function getCurrentType(): ?string
+    {
+        return $this->currentType;
+    }
+
+    public function getExpectedTypes(): ?array
```

Comment:
what about adding more type information when possible:
```suggestion
    /**
     * @return string[]|null
     */
    public function getExpectedTypes(): ?array
```

Reply:
Fixed, thanks